libsolidity/parsing/Token.cpp

```diff
@@ -50,6 +50,44 @@ namespace dev
 namespace solidity
 {
 
+bool ElementaryTypeNameToken::isElementaryTypeName(Token::Value _baseType, string const& _info)
+{
+	if (!Token::isElementaryTypeName(_baseType))
+		return false;
+	string baseType = Token::toString(_baseType);
+	if (baseType.find('M') == string::npos)
```

Comment:
Please don't rely on the string representation of the base type - it should only be for informational purposes.
I think it would be better to check every single possibility for `_baseType` and throw if it is different.